### DISCUSSION TITLE
Reverse for ' ' with arguments ' ' not found. 1 pattern(s) tried: ['(?P<pk>[0-9]+)$']

### DISCUSSION THREAD

**Discussion by legkoszkur** [community/community#21452] [Other Feature Feedback, Questions, & Ideas]
<p>I decide to make this post because i could not found answer to my problem on existing posts here  and on Stack Overflow. But finally i found it so i want to share it with you.</p>
<p>I got this error:</p>
<blockquote>
<p>Reverse for ‘article-view’ with arguments ‘(‘6’, ‘6’)’ not found. 1 pattern(s) tried: [’(?P[0-9]+)$’]</p>
</blockquote>
<p>during processing this line of code:</p>
<pre><code class="lang-auto"> def get_absolute_url(self):
         return reverse('article-view', args=(str(self.id)))
</code></pre>
<p>The problem was with brackets with should be square.</p>
<pre><code class="lang-auto">args=[str(self.id)]
</code></pre>
<p>You can see that on django website.</p>
<p><a href="https://docs.djangoproject.com/en/3.1/ref/urlresolvers/#django.urls.reverse" class="onebox" target="_blank" rel="noopener nofollow ugc">https://docs.djangoproject.com/en/3.1/ref/urlresolvers/#django.urls.reverse</a></p>

**Comment by am-giordano** [comment, marked as answer]
<p>The problem is that you first intended to pass a tuple to args, but you actually didn’t.</p>
<p>When creating a tuple with a single element you must add a comma after the element to indicate it is indeed a tuple. Otherwise the parenthesis would have the usual purpose of enclosing information (<code>x is (x)</code>).</p>
<p>Therefore, in your case you should use: <code>args=(str(self.id),)</code>. But of course your solution of passing a list instead is equally valid.</p>
